examples/flutter_gallery/lib/demo/pan_and_zoom_demo_transform_interaction.dart

```diff
@@ -0,0 +1,469 @@
+import 'package:flutter/material.dart';
+import 'package:vector_math/vector_math_64.dart' show Vector3;
+import 'pan_and_zoom_demo_inertial_motion.dart';
+
+// This widget allows 2D transform interaction on its child. The user can drag
+// to pan, pinch to zoom and rotate, and still get back untransformed
+// coordinates for targeted tap events.
+@immutable
+// TODO(justinmc): Need a better name. The plan is to make this a full blown
+// reusable Flutter widget.
+class TransformInteraction extends StatefulWidget {
+  const TransformInteraction({
+    // The child to perform the transformations on
+    @required this.child,
+    // The desired size of the area that can receive events
+    @required this.size,
+    // The scale will be clamped to between these values
+    this.maxScale = 2.5,
+    this.minScale = 0.8,
+    // Transforms will be limited so that the viewport can not view beyond this
+    // Rect. The Rect does not rotate with the rest of the scene, so it is
+    // always aligned with the viewport.
+    this.visibleRect,
+    // Initial values for the transform can be provided
+    this.initialTranslation,
+    this.initialScale,
+    this.initialRotation,
+    // Any and all of the possible transformations can be disabled.
+    this.disableTranslation = false,
+    this.disableScale = false,
+    this.disableRotation = false,
+    // If set to true, this widget will animate back to its initial transform
+    // and call onResetEnd when done. When utilizing reset, onResetEnd should
+    // also be implemented, and it should set reset to false when called.
+    this.reset = false,
+    // Access to event callbacks from GestureDetector. Called with untransformed
+    // coordinates in an Offset.
```

Comment:
The physics will need to be configurable once we make a reusable widget. Worth working through that mentally to make sure this class will be compatible.